test/test_runner/csv_to_parquet_converter.cpp

```diff
@@ -0,0 +1,126 @@
+#include "test_runner/csv_to_parquet_converter.h"
+
+#include <fstream>
+
+#include "common/file_utils.h"
+#include "common/string_utils.h"
+#include "spdlog/spdlog.h"
+#include "test_helper/test_helper.h"
+#include <arrow/csv/api.h>
+#include <arrow/io/api.h>
+#include <arrow/ipc/api.h>
+#include <parquet/arrow/reader.h>
+#include <parquet/arrow/writer.h>
+
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace testing {
+
+arrow::Status CSVToParquetConverter::RunCSVToParquetConversion(
+    const std::string& inputFile, const std::string& outputFile, char delimiter, bool hasHeader) {
+    std::shared_ptr<arrow::io::FileOutputStream> outFileStream;
+    std::shared_ptr<arrow::io::ReadableFile> infile;
+    std::shared_ptr<arrow::Table> csvTable;
+    ARROW_ASSIGN_OR_RAISE(infile, arrow::io::ReadableFile::Open(inputFile));
+    auto readOptions = arrow::csv::ReadOptions::Defaults();
+    auto parseOptions = arrow::csv::ParseOptions::Defaults();
```

Comment:
My understanding is that you are relying on arrow's auto csv reader to figure out data types for each csv column. I'm not sure if it's always as expected, especially when it comes to dates/timestamps/nested data types. I believe arrow converts some nested data types into strings. Can you double check the metadata of generated parquet files?
Ideally, we need to figure out data types for each column, and let the arrow reader be aware of that.
I'm fine with this auto reader for now if it works mostly as expected.

Actually I think the best way is we should add native support of exporting tables to parquet files in Cypher.
Then the conversion would be much easier.

Besides this, I'm curious what's the default row group size when we dump data into parquet?

Reply:
That's right, I'm relying on arrow's auto csv reader and it is not much assertive. Dates and timestamps are being converted into strings.